File: mediumSpringWebflux/src/main/java/com/example/mediumspringwebflux/domain/board/service/FindByIdBoard.java
package com.example.mediumspringwebflux.domain.board.service;

import com.example.mediumspringwebflux.domain.board.dto.BoardResponse;
import com.example.mediumspringwebflux.domain.board.service.facade.BoardToBoardResponse;
import com.example.mediumspringwebflux.domain.board.service.facade.FindBoard;
import org.springframework.stereotype.Service;
import reactor.core.publisher.Mono;

@Service
public class FindByIdBoard {
    private final BoardToBoardResponse boardToBoardResponse;
    private final FindBoard findBoard;

    public FindByIdBoard(BoardToBoardResponse boardToBoardResponse, FindBoard findBoard) {
        this.boardToBoardResponse = boardToBoardResponse;
        this.findBoard = findBoard;
    }

    public Mono<BoardResponse> execute(String id) {
        return Mono.defer(() -> findBoard.execute(id)
                .flatMap(boardToBoardResponse::rapping));
    }

}

File: mediumSpringWebfluxcorutine/src/main/kotlin/com/example/mediumspringwebfluxcorutine/board/service/FindByIdBoard.kt
package com.example.mediumspringwebfluxcorutine.board.service

import com.example.mediumspringwebfluxcorutine.board.dto.BoardResponse
import com.example.mediumspringwebfluxcorutine.board.service.facade.FindBoard
import org.springframework.stereotype.Service

@Service
class FindByIdBoard(
    private val findBoard: FindBoard
) {

    suspend fun execute(id: String): BoardResponse = findBoard.byId(id).toBoardResponse()

}